src/OsgViz.cpp

```diff
@@ -16,19 +16,24 @@
 namespace osgviz
 {
 
-osg::ref_ptr<OsgViz> instance = NULL;
 std::map<std::string, Module*> OsgViz::modules;
 
 osg::ref_ptr<OsgViz> OsgViz::getInstance(int argc,char** argv){
-	if (!instance.valid()){
-		instance = new OsgViz(argc,argv);
-	}
-	return instance;
+    //HACK to work around a static destruction order bug: default font needs to be initialized before
+    //     OsgViz. Calling getDefaultFont() here ensures that the font is instanciated before 
+    //     OsgViz and thus lives longer than OsgViz and can still be called during OsgViz destruction.
+    osgText::Font::getDefaultFont();
+    //NOTE it is important that instance is a local static variable because we need a way to 
+    //     control static initialization/destruction order.
+    //     If instance is global the static initialization order depends on the order in which 
+    //     libraries are loaded which leads to strange crashes in combination with osg's own singletons
+    static osg::ref_ptr<OsgViz> instance = new OsgViz(argc, argv);
```

Comment:
afaik this is the only way to control at what point in the execution order a singleton is created. That control is mandatory if we want to use OsgViz in a multi-library environment such as rock where other singletons might exist and singletons depend on each other.

In a good design singletons should never depend on each other but that is a design flaw in osg that we cannot fix :)